### PR TITLE
Introduce Object, FunctionTemplate

### PR DESCRIPTION
**Pull request by ksh8281**
* Add cctest for Object, FunctionTemplates
* Fix one of EncodedValue <-> EncodedSmallValue conversion bug

Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

File: src/runtime/FunctionTemplate.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+void* FunctionTemplate::operator new(size_t size)
+{
+    static bool typeInited = false;
+    static GC_descr descr;
+    if (!typeInited) {
+        GC_word objBitmap[GC_BITMAP_SIZE(FunctionTemplate)] = { 0 };
+        Template::fillGCDescriptor(objBitmap);
+        GC_set_bit(objBitmap, GC_WORD_OFFSET(FunctionTemplate, m_nativeFunctionData));
+        GC_set_bit(objBitmap, GC_WORD_OFFSET(FunctionTemplate, m_prototypeTemplate));
+        GC_set_bit(objBitmap, GC_WORD_OFFSET(FunctionTemplate, m_instanceTemplate));
```
Comment (clover2123) [inline src/runtime/FunctionTemplate.cpp:38]: member `Optional<FunctionTemplate*> m_parent` is omitted in this new operator.
Does this member not need memory tracking?

File: src/runtime/GlobalObject.cpp [hunk 2]
```diff
@@ -40,6 +40,9 @@
 #include "EnvironmentRecord.h"
 #include "Environment.h"
 
+#include "ObjectTemplate.h"
+#include "FunctionTemplate.h"
+
```
Comment (clover2123) [inline src/runtime/GlobalObject.cpp:45]: Why are these header files included? It seems that there is no usage in this cpp file.

File: build/escargot.cmake [hunk 3]
```diff
@@ -118,4 +124,12 @@ ELSEIF (${ESCARGOT_OUTPUT} STREQUAL "static_lib")
...
     TARGET_COMPILE_DEFINITIONS (${ESCARGOT_TARGET} PUBLIC ${ESCARGOT_DEFINITIONS})
     TARGET_COMPILE_OPTIONS (${ESCARGOT_TARGET} PUBLIC ${ESCARGOT_CXXFLAGS} ${CXXFLAGS_FROM_ENV})
+
+ELSEIF (${ESCARGOT_OUTPUT} STREQUAL "cctest")
+    ADD_EXECUTABLE ("cctest" ${ESCARGOT_SRC_LIST})
+
+    TARGET_LINK_LIBRARIES ("cctest" ${ESCARGOT_LIBRARIES} ${ESCARGOT_LDFLAGS} ${LDFLAGS_FROM_ENV} gtest)
+    TARGET_INCLUDE_DIRECTORIES ("cctest" PUBLIC ${ESCARGOT_INCDIRS})
+    TARGET_COMPILE_DEFINITIONS ("cctest" PUBLIC ${ESCARGOT_DEFINITIONS})
+    TARGET_COMPILE_OPTIONS ("cctest" PUBLIC ${ESCARGOT_CXXFLAGS} -I${ESCARGOT_ROOT}/third_party/googletest/googletest/include/ ${CXXFLAGS_FROM_ENV})
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply build/escargot.cmake]: I changed it

File: src/runtime/FunctionTemplate.cpp (hunk 1 above)
Comment (ksh8281) [reply src/runtime/FunctionTemplate.cpp:38]: Oops. I forget it

File: src/runtime/GlobalObject.cpp (hunk 2 above)
Comment (ksh8281) [reply src/runtime/GlobalObject.cpp]: Yes. I was for test purpose. I removed it

**Pull request merged**